### ISSUE TITLE
Data coverage, allele counts graphics and no calls

### ISSUE THREAD

**Issue by AlistairNWard** [tonydisera/gene.iobio#274]
I'm going to close a couple of older issues and wrap them into this. See the figure at the end for the observation. Demo exome dataset, TTN gene. Coverage of 2x, 7x, and 4x for proband, mother and father, but the father has no allele count graphic. Looking in the VCF file, the father is a no call, so there is no data for the father and he should have zero coverage. Perhaps the allele count graphic here should indicate 'no data' / 'no call'. The genotype field is essentially '.:.:.:.:.:.'.

![screen shot 2016-03-15 at 10 49 26 am](https://cloud.githubusercontent.com/assets/523111/13782276/d7a15d92-ea9c-11e5-8d6b-1a9aa39c40b2.png)

**Comment by AlistairNWard** [comment]
We should be handling no calls in a way that makes sense and so we could do a number of things:

If there is no genotype, we cannot assign one to it, but we should indicate that no call was made. Instead of the 'hom' or 'het' box, we could put 'no call'.

If we want to get the coverage at the exact location we can still do that directly from the bam file. e.g. the following command:

samtools view NA19238.bam 17:17644396-17644396 | wc -l

returns a list of all reads that intersect the specific position and then counts them. This returns 3, 0 and 2 for the 3 samples which agrees with the vcf records. So, we could still get the exact coverage for each sample for the mouse over, but we can't assign them to ref or alt without processing the actual read.

In conclusion - I would suggest that where there is a no call, we pull the coverage for the specific site from the bam, and then in the allele count put 'no call' in the box and a different coloured bar showing the allele count for unknown alleles.

**Comment by tonydisera** [comment]
This is a good idea.  The code already does an 'extra' annotation for an individual variant to get the hgvs notation.  So we could do the same thing when hovering over/clicking a variant.  When allele counts for a sample in the trio cannot be determined, perform the samtools your mentioned aboved to get the exact position.  Good thinking.

**Comment by tonydisera** [comment]
Should I try to fix this before our next release?

**Comment by AlistairNWard** [comment]
No. I think we just do everything totally necessary before getting to release and get the release to happen. This is a problem in the production anyway, so not addressing it immediately doesn't reduce functionality.

**Comment by tonydisera** [comment]
I agree.  I'll move it back to 'Ready' and we can re-prioritize after this release. :-)
